notebooks/metatranscriptomics_pipeline_iSeqnorm.ipynb

```diff
@@ -0,0 +1,1477 @@
+    "lanes = [1]\n",
+    "plate_df['vol_extracted_elution_ul'] = 100\n",
+    "\n",
+    "# Knight Lab Nextera is also valid for library_construction_protocol\n",
+    "metadata = {\n",
+    "    'Bioinformatics': [\n",
+    "        {\n",
+    "         'Sample_Project': 'NPH_15288',\n",
+    "         'QiitaID': '15288',\n",
+    "         'BarcodesAreRC': 'True',\n",
+    "         'ForwardAdapter': 'GATCGGAAGAGCACACGTCTGAACTCCAGTCAC',\n",
+    "         'ReverseAdapter': 'GATCGGAAGAGCGTCGTGTAGGGAAAGGAGTGT',\n",
+    "         'HumanFiltering': 'True',\n",
+    "         'library_construction_protocol': 'Knight Lab Kapa HyperPlus',\n",
+    "         'experiment_design_description': 'stool metatranscriptomics',\n",
+    "         'contains_replicates':plate_df['contains_replicates'].all(),\n",
+    "        },\n",
+    "    ],\n",
+    "    'Contact': [\n",
+    "        {\n",
+    "         'Sample_Project': 'NPH_15288',\n",
+    "         # non-admin contacts who want to know when the sequences\n",
+    "         # are available in Qiita\n",
+    "         'Email': 'klwetlab@health.ucsd.edu'\n",
+    "        },\n",
+    "    ],\n",
+    "    'Assay': 'Metatranscriptomic',\n",
+    "    'SheetType': 'standard_metat',\n",
+    "    'SheetVersion':'10'\n",
+    "}\n",
+    "\n",
+    "sheet = make_sample_sheet(metadata, plate_df, sequencer, lanes)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Write the sample sheet to file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Write the samplesheet as .csv\n",
+    "sample_sheet_fp = './test_output/SampleSheets/YYYY_MM_DD_NPH_3-6_metaT_samplesheet_iSeq_240.csv'\n",
+    "\n",
+    "if os.path.isfile(sample_sheet_fp):\n",
+    "    print(\"Warning! This file exists already.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open(sample_sheet_fp,'w') as f:\n",
+    "    sheet.write(f)\n",
+    "    \n",
+    "!head -n 30 {sample_sheet_fp}\n",
+    "!echo ...\n",
+    "!tail -n 15 {sample_sheet_fp}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "jupyter": {
+     "outputs_hidden": true
+    }
+   },
+   "source": [
+    "# Read Distribution Summary and Pool Normalization"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: import and merge per_sample read distributions\n",
+    "\n",
+    "Import a tsv file with read_counts from per_sample_fastq files and merge with growing plate_df\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#import reads counts from qiita prep-file file to dataframe\n",
+    "read_counts_df = pd.read_csv('./test_data/Demux/15288_prep_16673_20240401-173251.txt',\n",
```

Comment:
See above re Daniel's wish for dtype specification for read_csv() ...

Reply:
Also see my above comment re Daniel's suggestion being potentially tricky, and not worth doing unless you are very comfortable with how the data being read in is being used :)